staging/src/k8s.io/apiextensions-apiserver/pkg/openapi/construction.go

```diff
@@ -0,0 +1,327 @@
+							Patch:      c.patchOperation(Scale),
+							Parameters: pathParameters(),
+						},
+					},
+				},
+			},
+			Definitions: spec.Definitions{
+				model:     schema,
+				listModel: *c.listSchema(),
+				// TODO(roycaihw): this is a hack to let apiExtension apiserver and generic kube-apiserver
+				// to have the same io.k8s.api.autoscaling.v1.Scale definition, so that aggregator server won't
+				// detect name conflict and create a duplicate io.k8s.api.autoscaling.v1.Scale_V2 schema
+				// when aggregating the openapi spec. This hack won't tolerate any changes in
+				// io.k8s.api.autoscaling.v1.Scale (even any change in property description) because
+				// kube-openapi detect conflict using reflect.DeepEqual.
+				// We need to have a better fix.
+				"io.k8s.api.autoscaling.v1.Scale":       *scaleSchema(),
+				"io.k8s.api.autoscaling.v1.ScaleSpec":   *scaleSpecSchema(),
+				"io.k8s.api.autoscaling.v1.ScaleStatus": *scaleStatusSchema(),
+			},
+		},
+	}
+}
+
+// baseOperation initializes a base operation that all operations build upon
+func (c *SwaggerConstructor) baseOperation(kind ResourceKind, action string) *spec.Operation {
+	op := spec.NewOperation(c.operationID(kind, action)).
+		WithConsumes(
+			"application/json",
+			"application/yaml",
+		).
+		WithProduces(
+			"application/json",
+			"application/yaml",
+		).
+		WithTags(fmt.Sprintf("%s_%s", c.group, c.version)).
+		RespondsWith(401, unauthorizedResponse())
+	op.Schemes = []string{"https"}
+	op.AddExtension("x-kubernetes-action", action)
+
+	// Add x-kubernetes-group-version-kind extension
+	// For CRD scale subresource, the x-kubernetes-group-version-kind is autoscaling.v1.Scale
+	switch kind {
+	case Scale:
+		op.AddExtension("x-kubernetes-group-version-kind", []map[string]string{
+			{
+				"group":   "autoscaling",
+				"kind":    "Scale",
+				"version": "v1",
+			},
+		})
+	default:
+		op.AddExtension("x-kubernetes-group-version-kind", []map[string]string{
+			{
+				"group":   c.group,
+				"kind":    c.kind,
+				"version": c.version,
+			},
+		})
+	}
+	return op
+}
+
+// listOperation constructs a list operation for a CRD
+func (c *SwaggerConstructor) listOperation() *spec.Operation {
+	op := c.baseOperation(Resource, "list").
+		WithDescription(fmt.Sprintf("list or watch objects of kind %s", c.kind)).
+		RespondsWith(200, okResponse(fmt.Sprintf("#/definitions/%s.%s.%s", c.group, c.version, c.listKind)))
+	return addCollectionOperationParameters(op)
+}
+
+// createOperation constructs a create operation for a CRD
+func (c *SwaggerConstructor) createOperation() *spec.Operation {
+	ref := c.constructSchemaRef(Resource)
+	return c.baseOperation(Resource, "create").
+		WithDescription(fmt.Sprintf("create a %s", c.kind)).
+		RespondsWith(200, okResponse(ref)).
+		RespondsWith(201, createdResponse(ref)).
+		RespondsWith(202, acceptedResponse(ref)).
+		AddParam((&spec.Parameter{ParamProps: spec.ParamProps{Schema: spec.RefSchema(deleteOptionsSchemaRef)}}).
```

Comment:
Why would a create request have a deleteOptions in body?

Reply:
I must be copy-pasting code around. Fixed schema ref in create, replace and patch operations.